text/0000-relax-mixin-contract.md

```diff
@@ -0,0 +1,69 @@
+# Meta
+[meta]: #meta
+- Name: Relax Mixin Contract
+- Start Date: 2020-08-12
+- Author(s): @sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Mixins are used to imply multiple contracts:
+
+1. A contract between a buildpack and a stack, where the buildpack may state that it can only be used with stack images that have certain dependencies.
+
+2. A contract between a stack run image and stack build image, where apps built on the build image must have a certain dependency present on the run image to start successfully on the run image.
+
+Currently, if a stack build image has the mixin `build:mypkg` and a stack run image has the mixin `run:mypkg`, a buildpack cannot require the mixin as `mypkg` (either statically in buildpack.toml, or dynamically via the stackpack mechanism described in #87). A stack author may choose to use this configuration because the same dependency is used in different, unrelated ways in both images.
+
+Similarly, if a stack build image shares the mixin `mypkg` with the stack run image, a buildpack cannot require the mixin `build:mypkg` or `run:mypkg`, even though both images contain `mypkg`.
+
+This RFC proposed that we update the buildpack specification to create a new relationship between `mypkg`, `run:mypkg`, and `build:mypkg`.
+
+- While `run:mypkg` may imply a different set of changes to the run image compared to `mypkg`, buildpacks requiring `mypkg` should be satisfied with `run:mypkg` at runtime (and vice versa).
```

Comment:
I understand the differences between the two contracts conceptually but it would be helpful to have a realistic example where the run-image changeset described by `run:<pk>` vs `<pkg>` would differ.

Reply:
Removed differences instead of providing an example, as discussed above.